Fix test_worker_thread_count assertion for Python 3.13 (expects {21,22,23} but gets 20)

## Problem

In build 254, the `core: minimal tests 3.13 [g4_s13]` step fails because `test_worker_thread_count` in `python/ray/tests/test_basic.py` asserts `ray.get(actor.get_thread_count.remote()) in {21, 22, 23}` but gets `20` on Python 3.13.

The failure is deterministic — all 3 Bazel retry attempts fail identically: `assert 20 in {21, 22, 23}`.

This only fails on the `py3.13` variant of `g4_s13`. All other Python versions (3.10, 3.11, 3.12, 3.14) pass.

## Root Cause

Python 3.13 has one fewer background thread than previous versions in the Ray actor worker process, yielding thread count 20 instead of 21-23.

## Fix

In `python/ray/tests/test_basic.py` line 271, change:

```python
assert ray.get(actor.get_thread_count.remote()) in {21, 22, 23}
```

to:

```python
assert ray.get(actor.get_thread_count.remote()) in {20, 21, 22, 23}
```

The test comment at line 269-270 explicitly says:
> Lowering these numbers in this assert should be celebrated,  
> increasing these numbers should be scrutinized

A reduction from 21→20 threads is a positive change.

## Evidence

Build 254, step `g4_s13` (job `019ce764-be1e-4fbc-ab57-7fe177a49a14`):
```
FAILED python/ray/tests/test_basic.py::test_worker_thread_count - assert 20 in {21, 22, 23}
```
Failed in 3/3 retry attempts. 33 other tests in the same suite pass, 2 are skipped.

## Acceptance Criteria

- [ ] `python/ray/tests/test_basic.py` line 271 updated to include `20` in the expected set
- [ ] No other test files reference this assertion

--- a/python/ray/tests/test_basic.py
+++ b/python/ray/tests/test_basic.py
@@ -268,7 +268,7 @@ def test_worker_thread_count(monkeypatch, shutdown_only):
         ray.get(actor.get_thread_count.remote())
     # Lowering these numbers in this assert should be celebrated,
     # increasing these numbers should be scrutinized
-    assert ray.get(actor.get_thread_count.remote()) in {21, 22, 23}
+    assert ray.get(actor.get_thread_count.remote()) in {20, 21, 22, 23}
 
 
 # https://github.com/ray-project/ray/issues/7287

